src/PositionManager.sol

```diff
@@ -207,6 +211,12 @@ contract PositionManager is IPositionManager, ERC721Permit, PoolInitializer, Mul
         if (liquidity > 0) revert PositionMustBeEmpty();
     }
 
+    // implementation of abstract function DeltaResolver._pay
+    function _pay(Currency token, address payer, uint256 amount) internal override {
+        // TODO this should use Permit2
+        ERC20(Currency.unwrap(token)).safeTransferFrom(payer, address(poolManager), amount);
```

Comment:
wondering if we should also support `transfer` not just transferFrom?

Reply:
oh this was just a draft function to get posm to functional-parity with where it was before this PR. I'm assuming youll redo the `_pay` implementation to use Permit2.
As for whether posm needs `transfer` (universal router does). That depends if posm will ever custody funds that need to go into another position? I dont think so because you just leave them in pool manager and the deltas handle it all. So I would imagine posm only needs transferFrom unless you have a counter example?

Reply:
we may want to consider supporting transfer? but youre right we dont need it in this PR